src/System.Text.Json/tests/JsonNodeTests.JsonNumber.cs

```diff
@@ -0,0 +1,632 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static partial class JsonNodeTests
+    {
+        [Fact]
+        public static void TestDefaultCtor()
+        {
+            var jsonNumber = new JsonNumber();
+            Assert.Equal(0, jsonNumber.GetByte());
+            Assert.Equal(0, jsonNumber.GetInt16());
+            Assert.Equal(0, jsonNumber.GetInt32());
+            Assert.Equal(0, jsonNumber.GetInt64());
+            Assert.Equal(0, jsonNumber.GetSingle());
+            Assert.Equal(0, jsonNumber.GetDouble());
+            Assert.Equal(0, jsonNumber.GetSByte());
+            Assert.Equal((ushort)0, jsonNumber.GetUInt16());
+            Assert.Equal((uint)0, jsonNumber.GetUInt32());
+            Assert.Equal((ulong)0, jsonNumber.GetUInt64());
+            Assert.Equal(0, jsonNumber.GetDecimal());
+        }
+
+        [Theory]
+        [InlineData(17)]
+        [InlineData(byte.MinValue)]
+        [InlineData(byte.MaxValue)]
+        public static void TestByte(byte value)
+        {
+            // Default constructor:
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetByte(value);
+            Assert.Equal(value, jsonNumber.GetByte());
+            Assert.True(jsonNumber.TryGetByte(out byte result));
+            Assert.Equal(value, result);
+
+            // Numeric type constructor:
+            jsonNumber = new JsonNumber(value);
+            Assert.Equal(value, jsonNumber.GetByte());
+            Assert.True(jsonNumber.TryGetByte(out result));
+            Assert.Equal(value, result);
+
+            // Implicit cast:
+            jsonNumber = value;
+            Assert.Equal(value, jsonNumber.GetByte());
+            Assert.True(jsonNumber.TryGetByte(out result));
+            Assert.Equal(value, result);
+
+            // String constructor:
+            jsonNumber = new JsonNumber(value.ToString());
+            Assert.Equal(value, jsonNumber.GetByte());
+            Assert.True(jsonNumber.TryGetByte(out result));
+            Assert.Equal(value, result);
+        }
```

Comment:
What state are all of the other Get/TryGet methods supposed to be in? (If the "cross-type" tests are later, that's fine).